website/docs/reference/resource-properties/constraints.md

```diff
@@ -303,6 +303,67 @@ select
 
 </File>
 
+### Custom constraints on models for advanced configuration of tables
+
+In dbt Cloud, you can use custom constraints on models for advanced configuration of tables. Custom constraints allow you to set [masking policies](https://docs.snowflake.com/en/user-guide/security-column-intro#what-are-masking-policies) in Snowflake when using a  Create Table As Select (CTAS).
+
+Masking policies conceal sensitive data from unauthorized access, while still allowing authorized users to retrieve the data during query execution.
+
+Contracts and constraints support tag-based masking policies using the following syntax:
+
+<File name='models/constraints_example.sql'>
+
+```shell
+
+models:
+
+- name: my_model
+config:
+contract: {enforced: true}
+materialized: table
+columns:
+    - name: id
+    data_type: int
+    constraints:
+        - type: custom
+        expression: "tag (my_tag = 'my_value')" #  A custom SQL expression used to enforce a specific constraint on a column.
+
+```
+
+</File>
+
+Using this syntax requires defining all the columns and their types as it’s the only way where to send a create or replace `<cols_info_with_masking> mytable as ...`. It’s not possible to do it with just a partial list of columns.
```

Comment:
i personally don't fully understand what it means to 'defining all the columns and their type'  - could we have an example or be more specific? like for eg:
`Using this syntax requires configuring all the columns and their types as it’s the only way where to send a create or replace `<cols_info_with_masking> mytable as ...`. It’s not possible to do it with just a partial list of columns. This means making sure the `columns` and `constraints` fields are fully defined.`